File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem21_UniqueNumberOfOccurrences.java
package org.phoenix.leetcode.challenges;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class Problem21_UniqueNumberOfOccurrences {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int item : arr) {
            map.put(item, map.getOrDefault(item, 0) + 1);
        }
        Set<Integer> set = new HashSet<>(map.values());
        return map.size() == set.size();
    }
}

File: JavaScript/leetcode75/src/main/Problem21_UniqueNumberOfOccurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function (arr) {
  let map = new Map();
  for (let item of arr) {
    if (map.has(item)) {
      map.set(item, map.get(item) + 1);
    } else {
      map.set(item, 1);
    }
  }
  let set = new Set(map.values());

  return map.size === set.size;
};

module.exports = uniqueOccurrences;
